rust/automerge/src/op_set2/op.rs

```diff
@@ -262,12 +262,15 @@ impl TxOp {
         obj: ObjMeta,
         pos: usize,
         index: usize,
-        _action: types::OpType,
+        action: ResolvedAction,
         elemid: ElemId,
         pred: Vec<OpId>,
-        noop: bool,
     ) -> Self {
-        let (action, value, expand, mark_name) = _action.clone().decompose();
+        let (action, noop) = match action {
```

Comment:
how about `let (action, value, expand, mark_name, noop) = action.decompose();` and merge these 5 lines into 1 since thats all thats happening

Reply:
Also it looks like the `clone()` before decompose is not needed.  Great to remove that while where in here.